bukkit/latest/src/main/java/io/github/wysohn/triggerreactor/bukkit/manager/trigger/share/api/placeholder/PlaceholderExpansionSupport.java

```diff
@@ -93,15 +94,31 @@ public String getVersion() {
      */
     @Override
     public String onPlaceholderRequest(Player player, String identifier) {
-
-        //%tr_version% -> this should return TR version, but should use PluginDescription which is modified as protected method.
-
-
         if (identifier.toLowerCase().equals("version")) {
             return plugin.getVersion();
         }
+        //%tr_?<variable name>% - temporary global variable
+        if (identifier.substring(0, 1).equals("?")) {
+            String variableName = identifier.substring(1).replace('_', '.');
+            TemporaryGlobalVariableKey tempKey = new TemporaryGlobalVariableKey(variableName);
+            AbstractVariableManager.GlobalVariableAdapter adapter = (AbstractVariableManager.GlobalVariableAdapter) plugin.getVariableManager().getGlobalVariableAdapter();
+            Object value = adapter.get(tempKey);
+            if (value == null) {
+                return "";
+            }
+            if (value instanceof Number) {
+                value = String.valueOf(value);
+            }
+
+            if (!(value instanceof String)) {
+                return "";
+            } else {
+                String output = (String) value;
+                return output;
+            }
+        }
+
         // %tr_<variable name>%
-        //if(identifier.contains("")){return "";}
```

Comment:
I think this is what you're saying.  idk why I made this as comment.

Reply:
pretty close, yet empty String is empty String; I didn't mean to delete it

you can just check if the String's length is 0 or not.

I don't think PlaceholderAPI will pass anything other than empty String when no parameters are provided.

Reply:
@wysohn idk why I remove that line neither (thinking)

Reply:
@wysohn I think It's better to return empty string for these condition